### PR TITLE
Fix nxp 32872 flaky 2023

### PR DESCRIPTION
**Pull request by guirenard**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.nuxeo</groupId>
     <artifactId>nuxeo-parent</artifactId>
-    <version>2023.4.13</version>
+    <version>2023.14</version>
```
Comment (guirenard) [inline pom.xml:7]: I could have put this diff in a separate commit referencing [NXP-32743](https://jira.nuxeo.com/browse/NXP-32743)

Comment (kevinleturc) [reply pom.xml:7]: I think you can add it to your Align on 2023.14 commit if you want.

**Review by kevinleturc — APPROVED** [review]

**Review by ataillefer — APPROVED** [review]
Is https://jira.nuxeo.com/browse/NXP-32361 the right issue for the 3rd commit?

**Comment by guirenard** [comment]
> Is https://jira.nuxeo.com/browse/NXP-32361 the right issue for the 3rd commit?

Yes, I have spent some time incrementally upgrading the cold storage add-on with each 2023-HF version, starting from `2024.5` up to `2024.10`. I discovered that the unit tests started to fail because of https://github.com/nuxeo/nuxeo-lts/commit/ca755ba4da0d2b5cc11e11a4e88b4ec82e4f16e4#diff-d397d4395de989fbc9dba6f00004f4313866e78c59902d216c83993986ca4e32R841.

I introduced async behavior, and we need to add a couple of `nextTransaction` calls to our tests, especially when we create a test document. See https://github.com/nuxeo/nuxeo-coldstorage/pull/289/files#diff-b1590be41cd3ccc517ab1592207f6f4e81f817e701fb91957dfff377cac84f03R429
